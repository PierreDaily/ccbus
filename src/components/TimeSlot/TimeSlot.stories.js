import TimeSlot from "./index.vue";

// More on default export: https://storybook.js.org/docs/vue/writing-stories/introduction#default-export
export default {
  title: "TimeSlot",
  component: TimeSlot,
  // More on argTypes: https://storybook.js.org/docs/vue/api/argtypes
  argTypes: {
    isActive: { control: "boolean" },
    isFocus: { control: "boolean" },
    time: {
      control: { type: "select" },
      options: ["12:00 am", "0:00pm", undefined],
    },
    timeLeft: {
      control: { type: "select" },
      options: ["3 min", "13 min", undefined],
    },
    busLetter: { control: "text" },
  },
};

// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args) => ({
  // Components used in your story `template` are defined in the `components` object
  components: { TimeSlot },
  // The story's `args` need to be mapped into the template through the `setup()` method
  setup() {
    return { args };
  },
  // And then the `args` are bound to your component with `v-bind="args"`
  template: '<time-slot v-bind="args" />',
});

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
