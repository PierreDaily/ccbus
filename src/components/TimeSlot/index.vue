<template>
  <li
    :class="{ ['bg-grey-light']: !isActive, ['opacity-60']: !isActive }"
    class="shrink-0 flex items-center border-solid border-b border-grey-light h-[60px] justify-between px-6 py-0 text-center"
    style=""
    tabindex="0"
    ref="container"
    :style="{ color: isActive ? 'blue' : 'red' }"
  >
    <BusIcon :busLetter="busLetter" />
    <TimeDisplay :time="time" />
    <CountDown :timeLeft="timeLeft" />
  </li>
</template>

<script setup lang="ts">
import { onMounted, useTemplateRef, watchEffect } from "vue";
import BusIcon from "../BusIcon/index.vue";
import TimeDisplay from "../TimeDisplay/index.vue";
import CountDown from "../CountDown/index.vue";

const container = useTemplateRef("container");
const { busLetter, time, timeLeft, isActive, isFocus } = defineProps({
  busLetter: String,
  time: String,
  timeLeft: String,
  isActive: Boolean,
  isFocus: Boolean,
});
const emit = defineEmits(["scrollToActiveSlot"]);
onMounted(() => {
  if (isFocus && container.value) {
    emit("scrollToActiveSlot", container.value.offsetTop);
  }
});

watchEffect(() => {
  if (isFocus && container.value) {
    emit("scrollToActiveSlot", container.value.offsetTop);
  }
});
</script>
