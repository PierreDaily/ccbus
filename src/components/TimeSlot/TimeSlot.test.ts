import { mount } from "@vue/test-utils";
import TimeSlot from "./index.vue";

describe("TimeSlot component", () => {
  it("should mount", async () => {
    // arrange
    expect(TimeSlot).toBeTruthy();

    const wrapper = mount(TimeSlot, {
      props: {
        time: "12:00am",
        timeLeft: "2 min",
        isActive: false,
        isFocus: false,
      },
    });

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should have 'disable' class if isActive props is false", () => {
    // arrange

    const wrapper = mount(TimeSlot, {
      props: {
        time: "12:00am",
        timeLeft: "2 min",
        isActive: false,
        isFocus: false,
      },
    });

    //assert

    expect(wrapper.find("div").html()).toContain(
      'class="bg-grey-light opacity-60'
    );
  });

  it("should have 'disable' class if isActive props is false", () => {
    // arrange

    const wrapper = mount(TimeSlot, {
      props: {
        time: "12:00am",
        timeLeft: "2 min",
        isActive: true,
        isFocus: false,
      },
    });

    //assert

    expect(wrapper.find("div").html()).not.toContain(
      'class="class="bg-grey-light opacity-60'
    );
  });
});
