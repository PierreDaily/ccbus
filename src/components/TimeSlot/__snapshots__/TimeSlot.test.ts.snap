// Vitest Snapshot v1

exports[`TimeSlot component > should mount 1`] = `
"<li class=\\"disable\\" style=\\"color: red;\\" tabindex=\\"0\\" data-v-62a17b5e=\\"\\">
  <div id=\\"test\\" aria-label=\\"B Bus\\" class=\\"border border-solid border-grey-light h-[25px] w-[50px] rounded-sm overflow-hidden\\" data-v-62a17b5e=\\"\\">
    <div class=\\"items-center inline-flex justify-center w-1/2 h-full bg-bus-green border-r-grey-light border-r\\"><img src=\\"/src/assets/icons/bus.svg\\" /></div>
    <div class=\\"items-center inline-flex justify-center w-1/2 h-full align-top font-sans text-black-light\\" aria-labelledby=\\"test\\">B</div>
  </div>
  <div data-v-015bd910=\\"\\" data-v-62a17b5e=\\"\\"><img src=\\"/src/assets/icons/clock.svg\\" data-v-015bd910=\\"\\" /><span data-v-015bd910=\\"\\">12:00am</span></div>
  <div data-v-62a17b5e=\\"\\"><img class=\\"h-3 relative top-[-1px] left-[-2px] w-3 inline\\" aria-label=\\"coming in\\" src=\\"/src/assets/icons/rss.svg\\" /><span class=\\"text-black-light font-sans text-lg font-normal leading-5\\" style=\\"width: 100px;\\">2 min</span></div>
</li>"
`;
